src/main/java/com/cloudbees/jenkins/support/impl/RunningJobs.java

```diff
@@ -0,0 +1,58 @@
+package com.cloudbees.jenkins.support.impl;
+
+import com.cloudbees.jenkins.support.api.Component;
+import com.cloudbees.jenkins.support.api.Container;
+import com.cloudbees.jenkins.support.api.Content;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.model.Executor;
+import hudson.security.Permission;
+import jenkins.model.Jenkins;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Extension
+public class RunningJobs extends Component {
+
+    @NonNull
+    @Override
+    public Set<Permission> getRequiredPermissions() {
+        return Collections.singleton(Jenkins.ADMINISTER);
+    }
+
+    @Override
+    @NonNull
+    public String getDisplayName() {
+        return "Running Jobs";
+    }
+
+    @Override
+    public void addContents(@NonNull Container result) {
+        result.add(
+            new Content("nodes/master/running-jobs.txt") {
+                @Override
+                public void writeTo(OutputStream os) throws IOException {
+                    try (OutputStreamWriter osw = new OutputStreamWriter(os, StandardCharsets.UTF_8); BufferedWriter bw = new BufferedWriter(osw); PrintWriter out = new PrintWriter(bw)) {
+                        Optional.ofNullable(Jenkins.getInstanceOrNull())
+                            .ifPresent(jenkins -> Arrays.stream(jenkins.getComputers())
+                                .flatMap(computer -> computer.getExecutors().stream())
+                                .filter(Executor::isBusy)
+                                .collect(Collectors.toList())
+                                .forEach(executor -> Optional.ofNullable(executor.getCurrentExecutable())
+                                    .ifPresent(executable -> out.println(executable.getParent().getOwnerTask().getUrl()))));
```

Comment:
This does not make much sense. You are printing the job name when you could be printing the specific build. And there is no deduplication, and the format is URL-oriented which is less than readable. So if three builds are running for one branch of a multibranch project, you will see something like

```
job/some-repo/job/master/
job/some-repo/job/master/
job/some-repo/job/master/
```

rather than what you might expect

```
some-repo/master#112
some-repo/master#113
some-repo/master#114
```

Suggest, as I did before:

```suggestion
                                    .ifPresent(executable -> out.println(executable))));
```

Reply:
…though it looks like this will not currently print friendly content for `PlaceholderExecutable`s. Based on my reading of `Executable.toString` Javadoc, this would be a bug in `workflow-durable-task-step-plugin`, easy enough to fix.

Reply:
@jglick I don't understand what you are suggesting I do here.

Reply:
Resolved by filtering PlaceholderExecutables

```
.forEach(executor -> Optional.ofNullable(executor.getCurrentExecutable())
                                .filter(executable -> !(executable.getParent() instanceof ExecutorStepExecution.PlaceholderTask))
```